insights/core/dr.py

```diff
@@ -428,18 +428,29 @@ def get_registry_points(component):
         (set): A list of the registry points found.
     """
     reg_points = set()
-
+    # Process RegistryPoint spec of Specs
     if is_registry_point(component):
         reg_points.add(component)
-    else:
-        for dep in get_dependents(component):
+        return reg_points
+    # Process Component
+    dependencies = get_dependencies(component)
+    if dependencies:
+        for dep in dependencies:
             if is_registry_point(dep):
                 reg_points.add(dep)
             else:
-                dep_reg_pts = get_registry_points(dep)
+                dep_reg_pts = get_registry_points(dep, module=module)
                 if dep_reg_pts:
                     reg_points.update(dep_reg_pts)
-
+        if reg_points:
+            # Component must have (at least) one RegistryPoint, unless ->
+            return reg_points
+        # it's a spec of DefaultSpecs
+    # Process spec of DefaultSpecs
```

Comment:
Oops, I think the following process is incorrect, although allows passing a customized `module` to get the wanted module where the "RegistryPoints" are defined, there is no chance in the "dr.run" to modify it :(. It needs more research here.

Reply:
Hi @vdymna and @ryan-blakley - I modified the method again regarding the above issue. For `datasource`s, it needs to iterate  **UP** the `dependents`, but for `combiner/parsers` (high level components), it needs to iterate **DOWN** to the `dependencies`. Please help review it one more time. Thanks.